### PR TITLE
[RISCV] Add sifive-p470 processor

### PR DESCRIPTION
**Pull request by michaelmaitland**
This is an OOO core that has a vector unit. For more information see https://www.sifive.com/cores/performance-p450-470.

Use the existing P400 scheduler model. This model is missing accurate vector scheduling support, but it will be added in a follow up patch.

Other tunings can come in future patches too.

### CODE REVIEW

File: llvm/lib/Target/RISCV/RISCVFeatures.td [hunk 1]
```diff
@@ -1371,7 +1371,6 @@ def NoConditionalMoveFusion  : Predicate<"!Subtarget->hasConditionalMoveFusion()
 
 def TuneSiFive7 : SubtargetFeature<"sifive7", "RISCVProcFamily", "SiFive7",
                                    "SiFive 7-Series processors">;
-
```
Comment (wangpc-pp) [inline llvm/lib/Target/RISCV/RISCVFeatures.td:1374]: Removed by accident?

File: clang/test/Driver/riscv-cpus.c [hunk 2]
```diff
@@ -304,6 +304,16 @@
 // MCPU-SIFIVE-P450-SAME: "-target-feature" "+zbs"
 // MCPU-SIFIVE-P450-SAME: "-target-abi" "lp64d"
 
+// RUN: %clang -target riscv64 -### -c %s 2>&1 -mcpu=sifive-p470 | FileCheck -check-prefix=MCPU-SIFIVE-P470 %s
+// MCPU-SIFIVE-P470: "-target-cpu" "sifive-p470"
```
Comment (wangpc-pp) [inline clang/test/Driver/riscv-cpus.c:308]: One feature one line please. :-)

File: llvm/lib/Target/RISCV/RISCVProcessors.td [hunk 3]
```diff
@@ -266,11 +272,47 @@ def SIFIVE_P450 : RISCVProcessorModel<"sifive-p450", SiFiveP400Model,
...
                                        FeatureUnalignedScalarMem,
                                        FeatureUnalignedVectorMem],
-                                      [TuneNoDefaultUnroll,
-                                       TuneConditionalCompressedMoveFusion,
-                                       TuneLUIADDIFusion,
-                                       TuneAUIPCADDIFusion,
-                                       FeaturePostRAScheduler]>;
+                                      SiFiveP400TuneFeatures>;
+
+def SIFIVE_P470 : RISCVProcessorModel<"sifive-p470", SiFiveP400Model,
```
Comment (dtcxzyw) [inline llvm/lib/Target/RISCV/RISCVProcessors.td:277]: The data sheet says P470 is RVA22 compliant. Can we use `!listconcat(RVA22S64Features, ...)` here? See the processor definition for `SPACEMIT_X60` below.

Comment (topperc) [reply llvm/lib/Target/RISCV/RISCVProcessors.td:277]: What's the benefit of adding supervisor extensions? They aren't useable from applications. It just makes the ELF attribute very long.

Comment (michaelmaitland) [reply llvm/lib/Target/RISCV/RISCVProcessors.td:277]: I've updated it to use a profile list, without the supervisor extensions.

**Review by wangpc-pp — APPROVED** [review]
LGTM.

**Review by topperc — APPROVED** [review]
LGTM

**Review by dtcxzyw — APPROVED** [review]
LGTM.

**Comment by wangpc-pp** [comment]
> LLVM Buildbot has detected a new failure on builder `flang-x86_64-windows` running on `minipc-ryzen-win` while building `clang,llvm` at step 7 "test-build-unified-tree-check-flang".
> 
> Full details are available at: https://lab.llvm.org/buildbot/#/builders/166/builds/8
> Here is the relevant piece of the build log for the reference

What a long reflex arc!

**Comment by mshockwave** [comment]
> > LLVM Buildbot has detected a new failure on builder `flang-x86_64-windows` running on `minipc-ryzen-win` while building `clang,llvm` at step 7 "test-build-unified-tree-check-flang".
> > Full details are available at: https://lab.llvm.org/buildbot/#/builders/166/builds/8
> > Here is the relevant piece of the build log for the reference
> 
> What a long reflex arc!

As the icing on the top, I think that buildbot failure has nothing to do with this patch

**Pull request merged**